build_tools/benchmarks/run_benchmarks_on_android.py

```diff
@@ -281,6 +293,25 @@ def __run_capture(self, android_case_dir: pathlib.PurePosixPath,
     stdout_redirect = None if self.verbose else subprocess.DEVNULL
     execute_cmd(capture_cmd, verbose=self.verbose, stdout=stdout_redirect)
 
+  def __deduce_taskset_from_run_config(
+      self, run_config: iree_definitions.E2EModelRunConfig) -> str:
+    """Deduces the CPU mask according to device and execution config."""
+    exec_config = run_config.module_execution_config
+    single_thread = False
+    if exec_config.driver != iree_definitions.RuntimeDriver.LOCAL_TASK:
```

Comment:
Should this instead be looking for `LOCAL_SYNC`?

Reply:
The legacy logic also runs GPU benchmarks with the most performant core. Refactor to be more verbosely represent all the legacy cases, and fail if it falls outside those cases.